lib/src/chart/bar_chart/bar_chart_painter.dart

```diff
@@ -226,7 +226,20 @@ class BarChartPainter extends AxisChartPainter<BarChartData> with TouchHandler<B
                 bottomRight: borderRadius.bottomRight);
           }
 
-          _barPaint.color = barRod.backDrawRodData.color;
+          if (barRod.colors.length > 1) {

```

Comment:
This change is useless because we don't support gradient on backDrawRodData yet,
you can add the needed fields (such as colors, start, end, stops) in the backDrawRodData too, and change this code.